onadata/apps/logger/models/xform.py

```diff
@@ -1037,6 +1040,19 @@ def set_object_permissions(sender, instance=None, created=False, **kwargs):
         set_project_perms_to_xform(instance, instance.project)
 
 
+def send_message_on_save(sender, instance, created=False, **kwargs):
+    """
+    Signal handler send message on save
+    """
+    send_message(
+        instance_id=instance.id, target_id=instance.id,
+        target_type=XFORM, user=instance.created_by,
+        message_verb=FORM_UPDATED)
```

Comment:
Could we have one generic signal handler for this? something like:

```
def send_message(sender, instance, created=False, **kwargs):
    verb = FORM_UPDATED if isinstance(instance, XForm) else SUBMISSION_REVIEW
    target_type = XFORM if isinstance(instance, XForm) else SUBMISSION_REVIEW
    send_message(
        instance_id=instance.id, target_id=instance.id,
        target_type=target_type, user=instance.created_by,
        message_verb=verb)
```

Reply:
It would be nice to have, but the xform save won't be triggered on submission review, hence the condition won't hold.
Also, so far we have five other verbs: 
- SUBMISSION_REVIEW, 
- SUBMISSION_CREATED, 
- SUBMISSION_EDITED, 
- SUBMISSION_DELETED, 
- FORM_UPDATED.
These verbs are relevant for specific actions that are set in different signals or serializers.